src/main/java/io/cryostat/agent/remote/InvokeContext.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright The Cryostat Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.cryostat.agent.remote;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.management.ManagementFactory;
+import java.lang.management.MemoryMXBean;
+
+import javax.inject.Inject;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.sun.net.httpserver.HttpExchange;
+import org.apache.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class InvokeContext implements RemoteContext {
+
+    private final Logger log = LoggerFactory.getLogger(getClass());
+    private final ObjectMapper mapper;
+
+    @Inject
+    InvokeContext(ObjectMapper mapper) {
+        this.mapper = mapper;
+    }
+
+    @Override
+    public String path() {
+        return "/mbean-invoke/";
+    }
+
+    /*
+    * public <T> T invokeMBeanOperation(
+           String beanName,
+           String operation,
+           Object[] parameters,
+           String[] signature,
+           Class<T> returnType)
+    */
+
+    @Override
+    public void handle(HttpExchange exchange) throws IOException {
+        try {
+            String mtd = exchange.getRequestMethod();
+            switch (mtd) {
+                case "POST":
+                    try (InputStream body = exchange.getRequestBody()) {
+                        MBeanInvocationRequest req =
+                                mapper.readValue(body, MBeanInvocationRequest.class);
+                        if (req.beanName.equals(ManagementFactory.MEMORY_MXBEAN_NAME)) {
```

Comment:
Rather than having an if-else, or a `switch`, or other mechanism for enumerating beans, can't we do an MXBean lookup by name using the `req.beanName` and then pass the other `req` fields to `invoke`?

```java
var mbeanServer = ManagementFactory.getPlatformMBeanServer();
if (!req.isValid()) // respond HTTP 400 Bad Request
Object response = mbeanServer
    .invoke(ObjectName.getInstance(req.beanName), req.operation, req.parameters, req.signature);
// respond HTTP 200 with response body JSON, or HTTP 201 if response is null
```